preprocessor/src/lib.rs

```diff
@@ -7,135 +7,51 @@
 
 mod rotation;
 mod step;
-
-use beacon_api_client::{BlockId, Client, ClientTypes, Value, VersionedValue};
+use eth2::mixin::RequestAccept as _;
+use eth2::types::Accept;
+use eth2::BeaconNodeHttpClient;
 use eth_types::Spec;
-use ethereum_consensus_types::bls::BlsSignature;
-use ethereum_consensus_types::{
-    BeaconBlockHeader, BlsPublicKey, ByteVector, LightClientBootstrap, LightClientFinalityUpdate,
-    LightClientUpdateCapella, Root,
+use ethereum_types::{
+    EthSpec, FixedVector, LightClientFinalityUpdate, LightClientFinalityUpdateCapella,
+    LightClientFinalityUpdateDeneb, PublicKeyBytes,
 };
-
-use itertools::Itertools;
+use ethereum_types::{ForkVersionedResponse, LightClientUpdate};
 use lightclient_circuits::witness::{CommitteeUpdateArgs, SyncStepArgs};
 pub use rotation::*;
-use serde::{Deserialize, Serialize};
-use ssz_rs::{Node, Vector};
 pub use step::*;
+use url::Url;
 
-pub async fn get_light_client_update_at_period<S: Spec, C: ClientTypes>(
-    client: &Client<C>,
+pub async fn get_light_client_update_at_period<S: Spec, T: EthSpec>(
```

Comment:
consider having EthSpec as an associated type in Spec?

Reply:
Good idea